app/api/models/user.py

```diff
@@ -119,8 +120,8 @@ def add_models_to_namespace(api_namespace):
     "User registration model",
     {
         "name": fields.String(required=True, description="User name"),
-        "username": fields.String(required=True, description="User username"),
-        "password": fields.String(required=True, description="User password"),
+        "username": fields.String(required=False, description="User username"),
+        "password": fields.String(required=False, description="User password"),
```

Comment:
Yes. Actually, when we a user signs up using social login, we first try to find them in our database. If a pre-existing user is not found, we create a new one. To facilitate the creation of a new user, these fields were made optional. I did some testing and it works fine (because we have validation checks for them in any case) and all the unit tests passed.

If there is a better solution to this, please let me know. We also have an ongoing [discussion](https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/iOS/near/206469947) about this on zulip mentorship ios channel.

Reply:
Sounds good @yugantarjain thank you for your explanation!